.github/workflows/ci.yml

```diff
@@ -113,6 +113,58 @@ jobs:
       env:
         SLACK_WEBHOOK_URL: ${{ secrets.SLACK_ALERT_WEBHOOK_URL }} # required
 
+  # Run static analysis on the PR
+  static-analysis:
+    name:    "Static Analysis"
+    # We're using buildjet for this as it's very slow on Github's own runners
+    runs-on: buildjet-4vcpu-ubuntu-2204
+
+    # Skip any PR created by dependabot to avoid permission issues:
+    if:      (github.actor != 'dependabot[bot]')
+
+    permissions:
+      checks:        write
+      contents:      read
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: reviewdog/action-setup@8e48baae926e97848f0863ae248f3b08e089c81f
+        with:
+          reviewdog_version: latest
+
+      - name: Install jq
+        uses: dcarbone/install-jq-action@91d8da7268538e8a0ae0c8b72af44f1763228455
+
+      - name: Install semgrep
+        run:  |
+              python3 -m pip install semgrep
+              python3 -m pip install --upgrade requests
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+          cache:           false
+
+      - name: Install ci tools
+        run:  |
+              go install honnef.co/go/tools/cmd/staticcheck@latest
+              go install github.com/kisielk/errcheck@latest
+              go install github.com/gordonklaus/ineffassign@latest
+
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # fail the CI on any errors found on any line modified within the PR
+        # this means we'll ignore lines in those files
+        #
+        # If we want to fail on any errors found in files modified, change to "-filter-mode=file"
+        # If you want to fail on any errors found in the entire codebase, change to "-filter-mode=nofilter"
+        run:  reviewdog -fail-on-error -filter-mode=added -reporter=github-pr-review
```

Comment:
@eandre should we filter on the file changed rather than the line changed in the PR?

That way any existing issues in files will come up in the PR’s too and while that might be annoying to have to fix something unrelated to your change, at least it will act as a moviater to slowly fix the code base when we introduce new checks